db/migrate/20200727223040_add_ancestry_to_ems_folder.rb

```diff
@@ -0,0 +1,61 @@
+class AddAncestryToEmsFolder < ActiveRecord::Migration[5.2]
+  class EmsFolder < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+    has_many :all_relationships, :class_name => "AddAncestryToEmsFolder::Relationship", :dependent => :destroy, :as => :resource
+    include ActiveRecord::IdRegions
+  end
+
+  class Relationship < ActiveRecord::Base
+    belongs_to :ems_folder, :class_name => 'AddAncestryToEmsFolder::EmsFolder', :foreign_key => :ems_folder_id
+  end
+
+  def up
+    add_column :ems_folders, :ancestry, :string
+    add_index :ems_folders, :ancestry
+
+    id_range = EmsFolder.rails_sequence_range(EmsFolder.my_region_number)
+
+    say_with_time("set ems folder ancestry from existing genealogy relationship resource information") do
+      connection.execute(<<-SQL)
+      UPDATE ems_folders
+      SET ancestry = new_ancestry
+      FROM (
+        -- join to resources to convert rel.id to vm.id (it is in resource_id column)
+        -- ARRAY_AGG takes multiple rows and converts to an array
+        SELECT ancestor_resources_for_ems_folders.ems_folder_id AS ems_folder_id,
+        ARRAY_TO_STRING(ARRAY_AGG(res_rels.resource_id)::VARCHAR[], '/') AS new_ancestry
+        FROM (
+          -- get vm and associated ancestor rel_id
+          -- unnest converts an array to multiple rows, 1 ancestor per row
+          -- with ordinality, and order by relationships.indx make sure ancestors stay in the same order
+          SELECT ems_folders.id AS ems_folder_id, relationships.id AS rel_id
+          FROM ems_folders
+          JOIN relationships a_rels ON a_rels.resource_id = ems_folders.id
+          AND a_rels.relationship = 'genealogy'
+          AND a_rels.resource_type = 'EmsFolder'
+          LEFT JOIN LATERAL UNNEST(STRING_TO_ARRAY(a_rels.ancestry, '/')::BIGINT[])
+          WITH ORDINALITY AS relationships(id, indx) ON TRUE
+          WHERE ems_folders.id BETWEEN #{id_range.first} AND #{id_range.last}
+          ORDER BY ems_folders.id, relationships.indx
+        ) AS ancestor_resources_for_ems_folders
+        JOIN relationships res_rels ON res_rels.id = ancestor_resources_for_ems_folders.rel_id AND res_rels.relationship = 'genealogy'
+        GROUP BY ancestor_resources_for_ems_folders.ems_folder_id
+      ) AS new_ancestors_for_ems_folders
+      WHERE new_ancestors_for_ems_folders.ems_folder_id = ems_folders.id
+      SQL
+    end
+
+    Relationship.where(:relationship => 'genealogy', :resource_type => 'EmsFolder', :resource_id => EmsFolder.all.select(:id)).delete_all
+  end
+
+  def down
+    say_with_time("create relationship records from ems folder ancestry") do
+      folders_with_ancestry = EmsFolder.select { |folder| folder.ancestry.present? }
```

Comment:
probably not an issue but can we do this in sql?
I'm pretty sure this will be nil and never `""`

```suggestion
      folders_with_ancestry = EmsFolder.where.not(:ancestry => nil)
```

Reply:
commenting so I don't forget to make this change on the other two prs as well